assets/js/trivy.js

```diff
@@ -1,54 +1,124 @@
-jQuery(function () {
-  const columns = [
-    {
-      data: 'name',
-      createdCell: function (td, _cellData, rowData) {
-        $(td).html(`<a href="/components/${cleanColumnOutput(rowData.name)}">${cleanColumnOutput(rowData.name)}</a>`)
+class TrivyRenderer {
+  constructor(csrfToken, viewMode) {
+    this.csrfToken = csrfToken
+    this.viewMode = viewMode
+  }
+
+  post = async (url, body) => {
+    const response = await fetch(url, {
+      method: 'POST',
+      credentials: 'same-origin',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': this.csrfToken,
       },
+      body: JSON.stringify(body),
+    })
+    if (!response.ok) {
+      throw new Error(`There was a problem calling: ${url}`)
+    }
+    return response.json()
+  }
+
+  fetchMessages = async componentNames => {
+    const vulnerabilities = await this.post('/reports/trivy/data', { componentNames })
+    // console.log(vulnerabilities)
+    createTable({
+      id: 'trivyTable',
+      data: vulnerabilities,
+      orderColumn: 2,
+      orderType: 'desc',
+      columns,
+    })
+  }
+
+  start = async componentNames => {
+    await this.fetchMessages(componentNames)
+  }
+}
+
+jQuery(function () {
+  $('#updateProduct,#updateTeam,#updateServiceArea,#updateCustomComponentView').on('click', async e => {
+    // e.preventDefault(e)
+    // const productCode = document.getElementById('product').value
```

Comment:
think we probably should include the e.preventDefault but delete the other commented out code?

Reply:
Yep

Reply:
sorted with https://github.com/ministryofjustice/hmpps-developer-portal/pull/227